R/format_PEW_temp.R

```diff
@@ -0,0 +1,541 @@
+                  Tarsus = as.numeric(Tarsus),
+                  Mass = as.numeric(Mass),
+                  TarsusPartner = as.numeric(TarsusPartner),
+                  MassPartner = as.numeric(MassPartner),
+                  ClutchSize = as.integer(ClutchSize),
+                  DateEgg1 = as.Date(DateEgg1),
+                  HatchDateD0 = as.Date(HatchDateD0),
+                  NumberOfRingedChicks = as.integer(NumberOfRingedChicks),
+                  DateRingingChicks = as.Date(DateRingingChicks),
+                  NoOfChicksD3 = as.integer(NoOfChicksD3),
+                  BroodMassD3 = as.numeric(BroodMassD3),
+                  ObservationTimeH = as.numeric(ObservationTimeH),
+                  Species = "CYACAE",
+                  PopID = "PEW",
+                  NestboxID = tolower(Nest),
+                  BroodID = ifelse(Method %in% c("Catch adults nestbox", "ChickRinging"),
+                                   paste(Year, NestboxID, "PEW", sep = "_"), NA)) %>%
+    #### Rename variables to standardized format
+    dplyr::rename(IndvID = Id,
+                  Sex =  Seks,
+                  ObserverID = Measurer) %>%
+    #### Remove columns which we do not store in the standardized format
+    dplyr::select(-FeatherCollection ,
+                  -BreathRateTime50Breaths,
+                  -FeathersPartner,
+                  -BreathRatePartnerTime50Breaths,
+                  -BloodSample,
+                  -TimeBloodSample,
+                  -BloodSampleDuration,
+                  -VisitRateVisitsH,
+                  -VrPartner,
+                  -VisitsAlternated,
+                  -VisitsAlternatedPartner,
+                  -VisitsSync10,
+                  -ChickAgeOfBehavObserv,
+                  -MateStrategy,
+                  -Nest) %>%
+    #### Reorder columns
+    dplyr::select(BreedingSeason,
+                  Species,
+                  PopID,
+                  IndvID,
+                  Sex,
+                  Age,
+                  everything()) %>%
+    #### Remove observations without ID
+    dplyr::filter(!is.na(IndvID)) %>%
+    #### Details
+    dplyr::mutate(PartnerId = ifelse(PartnerId == "MetalRing_FemK89_2019",
+                                     "MetalRing_F2019_K89", PartnerId))
+
+
+  #### BROOD DATA
+  #### done, need check
+
+  message("Compiling brood information...")
+
+  Brood_data <- create_brood_PEW(data = pew_data)
+
+
+  #### CAPTURUE DATA
+  #### done
+
+  # AGE_CALCULATED SEEMS NOT CORRECT ?
+  # (maybe also because the Age_observed was not correctly assigned?)
+  #
+  # Liam:
+  #   Seems to me that there are some records that are not captures (i.e. they don't have a date)
+  #   but just include brood information. So, I would explicitly remove the records with no date
+  #   from Capture_data but still use them in Brood_data.
+  #   In other cases where there are multiple captures in a year (with dates)
+  #   these should be treated as legitimate multiple captures.
```

Comment:
Generally, yes I agree. CaptureDate is required in the Capture_data table. However, in this case it does seem like there's some useful capture information in the records without dates. e.g. in the first row of the data, we can assume that 11462816 was observed at some time during 2015 because it was classified as the female for nestbox 1. This type of observation still tells us something about things like the age of the individual at subsequent captures (we already know that it was an adult in 2015). So I would:

- Where an individual is observed as a brood parent and is not captured otherwise, include that record as a capture (give it a random date in 2015, maybe 01-01-2015, and just be explicit that this is a made-up date).
- Where an individual has a separate capture record with a date, ignore the records without dates.